intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R

```diff
@@ -0,0 +1,127 @@
+# Set up gene level statistics table for intro to R module
+
+# C. Savonen for ALSF - CCDL
+
+# 2020
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Establish base dir
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Establish R dir
+intro_r_dir <- file.path(root_dir, "intro-to-R-tidyverse")
+
+# Establish input and output directories
+data_dir <- file.path(intro_r_dir, "data")
+plots_dir <- file.path(output_dir, "plots")
+
+# Create these output directories if they don't exist
+if (!dir.exists(output_dir)) {
+  dir.create(output_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+# Install limma if not installed
+if (!("limma" %in% installed.packages())) {
+  # Install limma
+  BiocManager::install("limma", update = FALSE)
+}
+
+# Install the Human annotation package
+if (!("org.Hs.eg.db" %in% installed.packages())) {
+  # Install org.Hs.eg.db
+  BiocManager::install("org.Hs.eg.db", update = FALSE)
+}
+
+# Read in the dataset
+gene_df <- readr::read_tsv(file.path(input_dir, 
+                                     "GSE44971.tsv")) %>% 
+  tibble::column_to_rownames("Gene")
+
+# Read in metadata
+metadata <- readr::read_tsv(file.path(input_dir, 
+                                      "metadata_GSE44971.tsv")) %>%
+  dplyr::select(-which(apply(is.na(.), 2, all)), 
+                sex = refinebio_sex) %>% 
+  dplyr::mutate(
+    # Clean up character names
+    tissue = gsub(" ", "_", tolower(tissue)),
+    # Turn our testing variables in factors
+    #tissue = as.factor(tissue),
+    #sex = as.factor(sex)
+    )
+
+## Put samples' data and metadata in the same order
+# Make the data in the order of the metadata
+gene_df <- gene_df %>% 
+  dplyr::select(metadata$geo_accession)
+
+#This set up is based on the example 11.4 Estrogen Data: A 2x2 Factorial Experiment in 
+# https://bioc.ism.ac.jp/packages/2.9/bioc/vignettes/limma/inst/doc/usersguide.pdf
```

Comment:
I was following this set up, @jashapiro whether or not this is correct for the questions we are asking, I do not know.

Reply:
That is okay, but you'll note they are only interested in treatment effects at each time point, not differences between time points. A better option for this would be the setup in section 8.7 of that doc, probably mostly the one in the bottom of page 49. This would give the mean sex effect, mean tumor-normal effect, and the interaction effect.

Reply:
Okay. Great. I will model it after that one then. Thanks!